sdk/trace/src/main/java/io/opentelemetry/sdk/trace/TracerConfig.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.concurrent.Immutable;
+
+@AutoValue
+@Immutable
+public abstract class TracerConfig {
+
+  private static final TracerConfig DEFAULT_CONFIG =
+      new AutoValue_TracerConfig(/* enabled= */ true);
+
+  public static TracerConfig disabled() {
+    return new AutoValue_TracerConfig(/* enabled= */ false);
+  }
+
+  public static TracerConfig enabled() {
+    return DEFAULT_CONFIG;
```

Comment:
really minor: 

to improve debuggability and extensibility, it could make sense to have a 3-value state: default, enabled and disabled.

Then default state can be (in theory) configured. I assume distros might have different opinions on the default. Also I assume a lot of discussions on what default looks like in the spec.

Also we can identify conflicting cases: e.g. someone explicitly enabled and then disabled scope looks a bit error prone comparing to someone changed scope config from default to enabled.

Reply:
🤔 the 3-value state makes the behavior harder to understand for users. I think we'd want to come up with a concrete list of use cases, and confirm its not possible to accommodate them with a 2-value state before considering the 3-value state. Simpler is best, but may need to accept additionally complexity if use cases demand it.